### PR TITLE
Adding Prettier

### PR DESCRIPTION
**Pull request by tgriesser**
I know this doesn't do much for the major architectural things that need to be changed in the library, but hopefully it'll help with developer experience as we start to tackle those.

@elhigu and @wubzz since you have a better idea of what the outstanding PR's are, let me know when this would be best to merge, so there's less merge conflicts.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -66,22 +70,32 @@
     "babel-preset-es2015-loose",
     "rimraf"
   ],
+  "lint-staged": {
+    "*.{js,json}": ["prettier --write", "git add"]
```
Comment (wubzz) [inline package.json]: Nice! 🥇

File: .prettierrc [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
+  "singleQuote": true,
+  "trailingComma": "es5"
```
Comment (wubzz) [inline .prettierrc]: Not sure on the general convention of things but personally I would always opt for `arrowParens: "always"` as well, but of course it's personal preference. Just a thought!

**Comment by wubzz** [comment]
This will help a lot in terms of development! Consistent codebase is key.

As I wrote in gitter I think this can be merged at any time since existing PRs can be fixed using `eslint --fix` *(covers most cases)*.

If we really want to wait then perhaps 0.15 is a good target.

**Comment by elhigu** [comment]
If I understood correctly what @wubzz was suggesting in gitter chat, we could apply prettier straight away and then run prettier to those old PRs too... if that doesn't work, merging code by hand to the master is not so big deal for me either. I can do that manual work for every branch that I'm interested in 👍 

Anyways these PRs are currently on my list:

https://github.com/tgriesser/knex/pull/1390
https://github.com/tgriesser/knex/pull/2401
https://github.com/tgriesser/knex/pull/2449
https://github.com/tgriesser/knex/pull/2476

**Comment by elhigu** [comment]
> If we really want to wait then perhaps 0.15 is a good target.

I believe next version could be 0.15.0. 0.14 series is starting to be stable already (well I still need to add stress testing for mssql and oracle though).

**Pull request closed without merging**